dashboard-metrics-report-generator/src/main/scala/it/pagopa/interop/dashboardmetricsreportgenerator/Jobs.scala

```diff
@@ -105,21 +109,30 @@ object Jobs {
   def getTokensData(readModel: FileManager, config: TokensBucketConfiguration): Future[TokensData] = {
 
     def getIssueDate(token: String): Future[OffsetDateTime] =
-      Future(token.parseJson)
-        .flatMap(js => Future(js.asJsObject))
+      Future(token.parseJson.asJsObject)
         .flatMap(_.fields.get("issuedAt").toFuture(GenericError("Missing issuedAt field in token")))
         .flatMap {
-          case JsNumber(value) => Future.successful(value.toLong)
+          case JsNumber(value) => value.toLong.toOffsetDateTime.toFuture
           case _               => Future.failed(GenericError("issuedAt should be a number"))
         }
-        .flatMap(_.toOffsetDateTime.toFuture)
 
-    readModel
-      .getAllFiles(config.bucket)(config.basePath)
-      .map(_.values.flatMap(new String(_).split('\n')).toList)
-      .flatMap(tokens => Future.traverse(tokens)(getIssueDate))
-      .map { issueTimes =>
-        val twoWeeksAgo: OffsetDateTime = OffsetDateTime.now().minusDays(14)
+    def allTokensPaths(): Future[List[readModel.StorageFilePath]] =
+      readModel.listFiles(config.bucket)(config.basePath)
+
+    def getTokens(path: String): Future[List[String]] =
+      readModel.getFile(config.bucket)(path).map(bs => new String(bs).split('\n').toList)
+
+    val twoWeeksAgo: OffsetDateTime = OffsetDateTime.now().minusDays(14)
```

Comment:
I suggest to pass a timezone like UTC (or better, you can use the date time supplier) to avoid to be tied to the configuration of the machine we are running on